### PR TITLE
Feaster of Souls

### PR DESCRIPTION
**Pull request by hyresu**
Still needs the map editing

### CODE REVIEW

**Comment by joao-blip** [comment]
```
 /data/scripts/quests/feaster_of_souls/the_pale_worm/boss_lever.lua
 ..pts/quests/feaster_of_souls/the_pale_worm/boss_lever.lua:42: attempt to index local 'condition' (a nil value)
```

and

`
data/scripts/quests/feaster_of_souls/the_pale_worm/lantern.lua
 ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: attempt to index local 'condition' (a nil value)
stack traceback:
  [C]: in function '__index'
   ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: in main chunk
`

**Comment by joao-blip** [comment]
anything else from the src changes is bugging the character's healing

**Comment by Badeburu** [comment]
data/scripts/quests/feaster_of_souls/the_pale_worm/lantern.lua ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: attempt to index local 'condition' (a nil value) stack traceback: [C]: in function '__index' ...cripts/quests/feaster_of_souls/the_pale_worm

How i do this?

**Comment by Badeburu** [comment]
Where i found The spells of all monsters? Can u post?

**Comment by Juanzin93** [comment]
data/scripts/quests/feaster_of_souls/the_pale_worm/lantern.lua ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: attempt to index local 'condition' (a nil value) stack traceback: [C]: in function '__index' ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: in main chunk

 got the same error from above, fixed it by adding this to conditions.cpp 


		case CONDITION_HEX:
			return new ConditionHex(id, type, ticks, buff, subId); 

but hex isnt working

also, this code is not letting the player heal

	if (targetPlayer && damage.primary.type == COMBAT_HEALING) {
			damage.primary.value *= targetPlayer->getBuff(BUFF_HEALINGRECEIVED) / 100.;
		}

@hyresu

**Comment by Broodly** [comment]
Do u have the spell for monsters?

**Comment by Broodly** [comment]
Lua Script Error: [Scripts Interface]
C:\Users\leand\Desktop\Con lucas\otservbr-global-develop\data\scripts\quests\feaster_of_soul\the_pale_worm\boss_lever.lua
...ipts\quests\feaster_of_soul\the_pale_worm\boss_lever.lua:42: attempt to index local 'condition' (a nil value)
stack traceback:
        [C]: in function '__index'
        ...ipts\quests\feaster_of_soul\the_pale_worm\boss_lever.lua:42: in main chunk
> boss_lever.lua [error]
^

Lua Script Error: [Scripts Interface]
C:\Users\leand\Desktop\Con lucas\otservbr-global-develop\data\scripts\quests\feaster_of_soul\the_pale_worm\lantern.lua
...scripts\quests\feaster_of_soul\the_pale_worm\lantern.lua:7: attempt to index local 'condition' (a nil value)
stack traceback:
        [C]: in function '__index'
        ...scripts\quests\feaster_of_soul\the_pale_worm\lantern.lua:7: in main chunk
> lantern.lua [error]
^

I tested all the other stuff and its well done, keep it real @hyresu

**Comment by Badeburu** [comment]
the "brother" will is dead before the pale worm... and need update the room with damage, dont has damage per second with have players at the second floor

**Review by Bipbop69 — CHANGES_REQUESTED** [review]
/data/scripts/quests/feaster_of_souls/the_pale_worm/boss_lever.lua
 ..pts/quests/feaster_of_souls/the_pale_worm/boss_lever.lua:42: attempt to index local 'condition' (a nil value)
and

data/scripts/quests/feaster_of_souls/the_pale_worm/lantern.lua ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: attempt to index local 'condition' (a nil value) stack traceback: [C]: in function '__index' ...cripts/quests/feaster_of_souls/the_pale_worm/lantern.lua:7: in main chunk


anything else from the src changes is bugging the character's healing

File: src/game.cpp [hunk 1]
```diff
@@ -5564,6 +5564,10 @@ bool Game::combatChangeHealth(Creature* attacker, Creature* target, CombatDamage
 			}
 		}
 
+		if (targetPlayer && damage.primary.type == COMBAT_HEALING) {
+			damage.primary.value *= targetPlayer->getBuff(BUFF_HEALINGRECEIVED) / 100.;
```
Comment (patrickjmeurer) [inline src/game.cpp:5568]: This condition is breaking the healing system, no type of healing is working.

**Review by patrickjmeurer — CHANGES_REQUESTED** [review]

**Comment by gabrielscolt0** [comment]
> Where i found The spells of all monsters? Can u post?

i need the spells, so i can test dread maiden

**Comment by LIKEAHYENA** [comment]
> > Where i found The spells of all monsters? Can u post?
> 
> i need the spells, so i can test dread maiden

Which spells do you need?

**Comment by gabrielscolt0** [comment]
summonsthedreadmaiden, summonsunwelcome, maidenring, deathring1x1. The other spells I think I can create

**Comment by gabrielscolt0** [comment]
> > > Where i found The spells of all monsters? Can u post?
> > 
> > 
> > i need the spells, so i can test dread maiden
> 
> Which spells do you need?

summonsthedreadmaiden, summonsunwelcome, maidenring, deathring1x1. The other spells I think I can create

File: data/scripts/quests/feaster_of_souls/entrances.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
+local config = {
+	[14515] = {position = Position({x = 31914, y = 32354, z = 8})},
+	[14516] = {position = Position({x = 31914, y = 32354, z = 8})},
+	[14518] = {position = Position({x = 33613, y = 31415, z = 8})},
+	[14519] = {position = Position({x = 33613, y = 31415, z = 8})},
+	[14530] = {position = Position({x = 33649, y = 31445, z = 10})},
+	[14531] = {position = Position({x = 33613, y = 31415, z = 8})},
```
Comment (patrickjmeurer) [inline data/scripts/quests/feaster_of_souls/entrances.lua:7]: This position is incorrect, please correct to `Position({x = 33649, y = 31445, z = 10})`

**Comment by oualid6496** [comment]
The following monster spells are missing
image

Also there seems to be a mixup between the teleports to The Pale Worm and The Fear Feaster

**Comment by majestyotbr** [comment]
PR closed due to inactivity.

**Comment by Broodly** [comment]
Amigo porque dejo de funcionar algunos scripts de aqui?
....data\scripts\quests\feaster_of_soul\netheworld\mini_bosses.lua
[2021-08-07 21:12:11.837] [error] ...cripts\quests\feaster_of_soul\netheworld\mini_bosses.lua:3: attempt to index field 'Bosses' (a nil value)
stack traceback:
        [C]: in function '__index'
        ...cripts\quests\feaster_of_soul\netheworld\mini_bosses.lua:3: in main chunk

Me esta saliendo eso, antes me funcionaba un par de update atras.. ahora ya no si alguien puede ayudarme

**Pull request closed without merging**